app/pages/sections/catalogue-solutions/select-price/contextCreator.js

```diff
@@ -1,8 +1,33 @@
 import manifest from './manifest.json';
 import { baseUrl } from '../../../../config';
 
-export const getContext = ({ orderId }) => ({
+const generatePriceList = solutionPricingData => solutionPricingData.prices.map((mappedPrice) => {
+  const timeUnitdescription = (mappedPrice.timeUnit || {}).description ? mappedPrice.timeUnit.description : '';
+  if (mappedPrice.type === 'flat') {
+    return {
+      value: `£${mappedPrice.price} ${mappedPrice.itemUnit.description} ${timeUnitdescription}`,
+      text: `£${mappedPrice.price} ${mappedPrice.itemUnit.description} ${timeUnitdescription}`,
+    };
+  }
+  let tieredHtml = '';
+  mappedPrice.tiers.forEach((tier) => {
+    const tieredRange = tier.end ? `${tier.start} - ${tier.end}` : `${tier.start}+`;
+    tieredHtml += `<div>${tieredRange} ${mappedPrice.itemUnit.tierName} £${tier.price} ${mappedPrice.itemUnit.description} ${timeUnitdescription}</div>`;
+  });
+  return {
+    value: tieredHtml,
+    html: tieredHtml,
+  };
```

Comment:
Just to make this bit more easier to read. I would extract these into 2 little helper functions. One to generate the flat price and the other to generate the tiered prices.